<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DialogProtocolSettings</class>
 <widget class="QDialog" name="DialogProtocolSettings">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>500</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Protocol Settings</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="protocol_quic_settings">
      <attribute name="title">
       <string notr="true">QUIC</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="1" column="0">
        <widget class="QLabel" name="down_l">
         <property name="text">
          <string>Download (Mbps)</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="up_l">
         <property name="text">
          <string>Upload (Mbps)</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLineEdit" name="down"/>
       </item>
       <item row="0" column="1">
        <widget class="MyLineEdit" name="up"/>
       </item>
       <item row="2" column="0">
        <widget class="QCheckBox" name="hy2speed">
         <property name="text">
          <string>Apply Speed Settings for Hysteria2</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="protocol_ech_settings">
      <attribute name="title">
       <string notr="true">VMess</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="2" column="0">
        <widget class="QCheckBox" name="global_padding">
         <property name="text">
          <string>Global Padding</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QCheckBox" name="authenticated_length">
         <property name="text">
          <string>Authenticated Length</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QComboBox" name="vmess_security">
         <item>
          <property name="text">
           <string notr="true">auto</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">chacha20-poly1305</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">aes-128-gcm</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="security_l">
         <property name="text">
          <string>Security</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="packet_encoding_l">
         <property name="toolTip">
          <string>Also for VLESS</string>
         </property>
         <property name="text">
          <string>Packet Encoding</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QComboBox" name="packet_encoding">
         <item>
          <property name="text">
           <string notr="true"/>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">packetaddr</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string notr="true">xudp</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyLineEdit</class>
   <extends>QLineEdit</extends>
   <header>ui/widget/MyLineEdit.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DialogProtocolSettings</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>227</x>
     <y>497</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DialogProtocolSettings</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>295</x>
     <y>503</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <designerdata>
  <property name="gridDeltaX">
   <number>10</number>
  </property>
  <property name="gridDeltaY">
   <number>10</number>
  </property>
  <property name="gridSnapX">
   <bool>true</bool>
  </property>
  <property name="gridSnapY">
   <bool>true</bool>
  </property>
  <property name="gridVisible">
   <bool>true</bool>
  </property>
 </designerdata>
</ui>
